### ISSUE TITLE
some cities don't have suggested by user

### ISSUE THREAD

**Issue by mehulkar** [TeaWithStrangers/tws-on-rails#360]
- we should make sure city suggestion is capturing a user
- see if log files have any info to be able to associate legacy suggested cities with a user
- make sure there isn't a bug with how UI is displaying suggestors

**Comment by ankitshah811** [comment]
just observed that there are two current flows. as follows (i think this is why we have no-user city suggestions):

**FLOW 1 — leads to submitted user being counted**
user goes to sign up page
user enters information and signs up
user is taken to city page
user doesn't see their city
user clicks 'add your city'
user enters information and is marked as the submitter

**FLOW 2 — leads to submitted user not counting**
user goes to cities page
user doesn't see their city
user clicks 'add your city'
user enters information and is then taken to sign up page [CITY IS CREATED HERE]
user enters information and signs up
user sees city page [is now confused and either stops because they figure they're suggested city is now the city they suggested, but is not!]
user *might* click 'add your city'
user enters information and is marked as the submitter [SECOND CITY IS CREATED HERE AND USER IS MARKED AS SUBMITTER]

**Comment by mehulkar** [comment]
Yeah, the two flows are confusing. We should figure out 1 flow and remove the other. 

I like this flow from a user perspective:

1. Non-signed-in user clicks "Let's get tea" from home page
2. Non-signed-in user is directed to /cities
3. Non-signed-in user looks at list of cities and doesn't see the one they want
4. Non-signed-in user clicks "Don't see your city? Add it here"
5. Non-signed-in user submits their city info
6. Non-signed-in user gets to thank you page (the unapproved city page)
7. Non-signed-in user is prompted to get notified if the city is approved
8. Non-signed-in user signs up
9. User is now a signed in user

This is bad from a product perspective because we aren't trying to get their info. But it's good from a user perspective because they don't have to sign up until it really makes sense for them to sign up and they _want_ to sign up (for their own city)

**Comment by ankitshah811** [comment]
I think we lose more users here than just the ones who would create new cities. Giving people lots to think about on `/cities` with # of options could make them less likely to sign up. This includes users in cities we're in and cities we're not. 

I think we can very fairly assume that a user creating a new city isn't doing it just because they want random cities to be on the page. They're doing it because they want to be signed up for that city (or — and this is as far as my mind stretches on this — they often travel between two cities and want their secondary city there). 

New proposal:

1. Non-signed-in user clicks "Let's get tea" from home page
2. Non-signed-in user is directed to /signup
3. Non-signed-in user signs up
4. Signed-in user looks at list of cities and doesn't see the one they want
5. User clicks "Don't see your city? Add it here"
6. User submits their city info
7. User gets to thank you page (the unapproved city confirmation page — http://cl.ly/image/3I0A451B3f1l).
8. User is now a signed in user with suggested city as theirs
  * If city is approved, this is maintained.
  * If city is declined, they have no city.
  * If city is merged into another one, their city changes to the one their suggestion was merged into

OR

1. Non-signed-in user goes to `/cities`
2. Non-signed-in user looks at list of cities and doesn't see the one they want
3. Non-signed-in user clicks "Don't see your city? Add it here"
4. Non-signed-in user is taken to `/signup` with a flash banner that says "Sign up before submitting a city so we can notify you when it's approved!"
5. Non-signed-in user signs up and is redirected to `/cities/submit`
6. Signed-in user submits their city info
7. User gets to thank you page (the unapproved city confirmation page — http://cl.ly/image/3I0A451B3f1l).
8. User is now a signed in user with suggested city as theirs
  * If city is approved, this is maintained.
  * If city is declined, they have no city.
  * If city is merged into another one, their city changes to the one their suggestion was merged into

**Comment by mehulkar** [comment]
This makes sense and it makes it so you have to be signed in to suggest a city. I think that's suboptimal for users, but it's also more in line with the rest of the city. I am on board with it as an incremental improvement.

TLDR:

When a user goes to /cities and clicks "add city", if they aren't sign in, it should go to sign up page instead of suggest page. 

Optional: in this particular flow, after signing up, they should go back to /cities/suggest. I'm making this optional, because this is a feature that should exist across the site (redirect to the place you were in before you were redirected to sign up or log in). There are solutions to this, but I don't think they're in the site anywhere yet, so we should not do this as a one-off thing. Moving to #363

**Comment by ankitshah811** [comment]
@mehulkar closing this = changes have been made and pushed?

**Comment by mehulkar** [comment]
adoption as home city doesn't really make sense as a feature, but I didn't make any changes on that. If it happened currently, it still does. If it didn't then it does not. Open a new issue to discuss/design that feature!

a not signed up user that tried to submit a city will be redirected to `/signup` not `/signin`. (sign up page, not login page). That has been done and merged in #364. Not deployed to production yet.

**Comment by ankitshah811** [comment]
@mehulkar flash banner?
